src/joystick/keyboard_control.cpp

```diff
@@ -41,34 +41,81 @@ int main(int argc, char **argv)
     while(ros::ok())
     {
         robosub::control msg = base_msg;
-        uint8_t key = getKey();
+        uint8_t key = getKey(), k1, k2;
         tcflush(fileno(stdin), TCIFLUSH);
         switch (key)
         {
             case 'w':
-                  msg.forward = 1;
-                  break;
+                msg.forward = 1;
+                break;
             case 's':
-                  msg.forward = -1;
-                  break;
+                msg.forward = -1;
+                break;
             case 'a':
-                 msg.strafe_left = 1;
-                 break;
+                msg.strafe_left = 1;
+                break;
             case 'd':
-                 msg.strafe_left = -1;
-                 break;
+                msg.strafe_left = -1;
+                break;
             case 'q':
-                 msg.yaw_left = 10;
-                 break;
+                msg.yaw_left = 10;
+                break;
             case 'e':
-                 msg.yaw_left = -10;
-                 break;
+                msg.yaw_left = -10;
+                break;
             case 'z':
-                 msg.dive = -0.25;
-                 break;
+                msg.dive = -0.25;
+                break;
             case 'c':
-                  msg.dive = 0.25;
-                  break;
+                msg.dive = 0.25;
+                break;
+            case 27:  // Escape Code for special keys
+                k1 = getKey(); // Get second code
+                k2 = getKey(); // Get third code
+
+                if ('[' == k1)  // Ctrl is not pressed
+                {
+                    switch(k2)
+                    {
+                        case 'A':  // Up Key pressed
+                            msg.forward = 1;
+                            break;
+                        case 'B':  // Down Key pressed
+                            msg.forward = -1;
+                            break;
+                        case 'C':  // Right key pressed
+                            msg.strafe_left = -1;
+                            break;
+                        case 'D':  // Left key pressed
+                            msg.strafe_left = 1;
+                            break;
+                        case '5':  // Page up Key pressed
+                            msg.dive = 0.25;
+                            break;
+                        case '6':  // Page down Key pressed
+                            msg.dive = -0.25;
+                            break;
+                    }
+                }
+                else if ('O' == k1)  // Ctrl is pressed
+                {
+                    switch(k2)
+                    {
+                        case 'A':  // Up Key pressed
+                            msg.pitch_down = 10;
+                            break;
+                        case 'B':  // Down Key pressed
+                            msg.pitch_down = -10;
```

Comment:
If the down key is meant to be pitch down, this should be positive. Positive `pitch_down` is pitching down. Also, pitch is absolute above. This would only allow for pitching up and down 10 degrees maximum.